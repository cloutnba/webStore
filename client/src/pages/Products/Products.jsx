import FilterMainList from './components/FilterMainList';
import { Container } from '@mui/material';
import {
  selectorAllProducts,
  selectorSearchInputValue,
  selectorServerErrorProducts,
  selectorProductsQuantity,
  selectorPageLoading,
  selectorFilterRequest,
  selectorUrlAddress,
  selectorShowPaginaton,
} from '../../selectors';
import {
  actionFetchAllProducts,
  actionFetchSearchFilterProducts,
  actionFetchSearchProducts,
  actionFilterRequest,
  actionFirstVisitAndResetToCorectFilter,
} from '../../reducers';
import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import ProductCard from '../../components/ProductCard';
import BreadCrumbs from '../../components/BreadCrumbs';
import ServerError from '../../components/Notifications/ServerError';
import Paginate from './components/Paginate';
import './Products.scss';
import Preloader from '../../components/Preloader';
import { useLocation } from 'react-router-dom';
import { useSearchParams } from 'react-router-dom';
import Button from '../../components/Button';
import {actionResetSearch} from "../../reducers/products.reducer";

const Products = () => {
  const allProducts = useSelector(selectorAllProducts);
  const productsQuantity = useSelector(selectorProductsQuantity);
  const searchInputValue = useSelector(selectorSearchInputValue);
  const serverError = useSelector(selectorServerErrorProducts);
  const pageLoading = useSelector(selectorPageLoading);
  const showPagination = useSelector(selectorShowPaginaton)


  const dispatch = useDispatch();

  const [searchParams, setSearchParams] = useSearchParams();
  const location = useLocation();
  let urlString = location.search
  let newFilterRequestObj = useSelector(selectorFilterRequest)
  let requestObj = { ...newFilterRequestObj }

  const urlAddress = useSelector(selectorUrlAddress) || urlString

  let getLinkObj = {}
  let stringToarr = urlString.substring(1)
  stringToarr = stringToarr.replaceAll(/%2C/ig, ',')
  stringToarr = stringToarr.replaceAll(/\+/g, " ")

  let arr = stringToarr.split('&')
  let arrKeyValue = arr.map((el) => {
    return el.split('=')
  })

  arrKeyValue.forEach((e) => {
    getLinkObj[e[0]] = e[1]
  })


  for (let key in requestObj) {
    for (let keyLink in getLinkObj) {
      if (key == keyLink) {
        if (key == 'perPage' || key == 'startPage') {
          requestObj[key] = Number(getLinkObj[keyLink])
        } else {
          requestObj[key] = getLinkObj[keyLink]
        }

      }
    }
  }

  useEffect(() => {
    setSearchParams(urlAddress)
  }, [urlAddress])

  const resetFilters = () => {
    dispatch(actionFirstVisitAndResetToCorectFilter(false))
    dispatch(actionFetchSearchFilterProducts({
      brand: '',
      category: '',
      processorBrand: '',
      screenSize: '',
      color: '',
      ramMemory: '',
      hardDriveCapacity: '',
      perPage: 6,
      startPage: 1,
      minPrice: '',
      maxPrice: '',
      sort: '',
    }))  

  }



  useEffect(() => {
    if (searchInputValue === '') {
      let obj = JSON.parse(sessionStorage.getItem('filterRequest'));
      if (obj) {
        dispatch(actionFetchSearchFilterProducts(obj));
      }
      else {
        dispatch(actionFilterRequest(requestObj))
        dispatch(actionFetchAllProducts(urlString));
   
      }
    } else {
      dispatch(actionFetchSearchProducts(searchInputValue));
    }
    return () => {

      dispatch(actionResetSearch());
    }
  }, [location]);

  return (
    <main>
      <Container className="main-list" maxWidth="lg">
        <Preloader open={pageLoading} />
        {serverError && <ServerError />}
        {!serverError && (
          <>
            <BreadCrumbs linksArray={[{ link: '/products', text: 'Products' }]} />
            <div className='products-header'>
              <p className="count-found-product">
                Products <span className="title_contrast">{productsQuantity} found</span>
              </p>
              <Button type={"button"} text="Reset filters" onClick={() => resetFilters()} className="btn-reset-filter" />

            </div>
            <section className="main-list__sections">
              <div className="main-list__sections--products">
                {allProducts.length > 0 && !serverError ? (
                  <>
                    <div className="grid-main-list">
                      {allProducts?.map((el, index) => (
                        <ProductCard el={el} key={el._id} index={index} />
                      ))}
                    </div>
                    {showPagination && <Paginate />}
                  </>
                ) : (
                  <p className="text-product__not-found">
                   {!pageLoading && 'Nothing to find, please enter correct name or change your filter'}
                  </p>
                )}
              </div>
              <div className="main-list__sections--filters">
                <FilterMainList />
              </div>
            </section>
          </>
        )}
      </Container>
    </main>
  );
};
export default Products;
