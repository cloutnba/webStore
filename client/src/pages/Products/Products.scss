@use './src/styles/mixins';

.main-list {
  position: relative;

  &__sections {
    display: grid;
    grid-template-columns: 6fr 1fr;
    column-gap: 20px;

    @media (max-width: 570px) {
      &--products{
        order: 2;
      }

      &--filters{
        order: 1;
      }

      display: flex;
      flex-direction: column;
    }
  }
}

.grid-main-list {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
 /*  grid-template-rows: repeat(1, 340px); */
 grid-auto-rows: 325px;
  gap: 30px;
  margin-bottom: 30px;

  @media (max-width: 1150px) {
    grid-template-columns: repeat(2, 1fr);
    gap: 30px;
    justify-items: center;
  }

  @media (max-width: 880px) {
    display: grid;
    grid-template-columns: repeat(1, 1fr);
    align-items: center;
    justify-content: center;
    align-content: center;
    justify-items: center;
    gap: 40px;
  }

}

.count-found-product {
  @include mixins.titleStyles;

  .title_contrast {
    @include mixins.liner-gradient-tile;
  }
}

.text-product__not-found {
  text-align: center;
  font-family: Ubuntu;
  font-style: normal;
  font-weight: 700;
  font-size: 26px;
}

.products-header{
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 30px;

}

.btn-reset-filter{
  width: 150px !important;
  border-radius: 10px !important;
  height: 37px !important;
}
