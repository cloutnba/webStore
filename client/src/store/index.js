import { configureStore } from '@reduxjs/toolkit';

import {
    productsReducer,
    favoritesReducer,
    scalesReducer,
    basketReducer,
    logInReducer,
    productDetailsReducer,
    discountedProductsReducer,
    checkoutReducer,
    personalOfficeReducer,
    registrationReducer
} from "../reducers";


const store = configureStore({
    reducer:{
        products: productsReducer,
        productsDetails: productDetailsReducer,
        favorites: favoritesReducer,
        scales: scalesReducer,
        basket: basketReducer,
        logIn: logInReducer,
        discountedProducts: discountedProductsReducer,
        checkout: checkoutReducer,
        personalOffice: personalOfficeReducer,
        registration: registrationReducer,
    }

})

export default store;
