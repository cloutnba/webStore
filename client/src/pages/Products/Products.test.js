import Products from "./Products";
import {render, screen} from "@testing-library/react"
import {BrowserRouter} from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../store'
import '@testing-library/jest-dom/extend-expect';

describe("Test Products", ()=>{
    test("snapshot Products", ()=>{
        const products = render(<Provider store={store}><BrowserRouter><Products/></BrowserRouter></Provider>) 
        expect (products).toMatchSnapshot() 
    });

    test("test container", ()=>{
        const {container} = render(<Provider store={store}><BrowserRouter><Products/></BrowserRouter></Provider>)
        const block = container.querySelector("section")
        expect(block).toHaveClass('main-list__sections')
    })

    test('check props', ()=>{
        const {container} = render(<Provider store={store}><BrowserRouter><Products serverError={false}/></BrowserRouter></Provider>)
        const productsList = container.querySelector(".notification__wrapper")
        expect(productsList).not.toBeInTheDocument();
    })

})