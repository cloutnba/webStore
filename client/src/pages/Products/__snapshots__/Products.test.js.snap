// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Products snapshot Products 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg main-list css-1oqqzyl-MuiContainer-root"
        >
          <div>
            <div
              aria-hidden="true"
              class="MuiBackdrop-root css-1mgwyik-MuiBackdrop-root"
              style="opacity: 0.7; background-color: white; position: absolute; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            >
              <span
                class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
                role="progressbar"
                style="width: 40px; height: 40px;"
              >
                <svg
                  class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                    cx="44"
                    cy="44"
                    fill="none"
                    r="20.2"
                    stroke-width="3.6"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="breadcrumbs__wrapper"
          >
            <a
              class="breadcrumbs__link"
              href="/"
            >
              Home
            </a>
            <a
              class="breadcrumbs__link"
              href="/products"
            >
              Products
            </a>
          </div>
          <div
            class="products-header"
          >
            <p
              class="count-found-product"
            >
              Products 
              <span
                class="title_contrast"
              >
                0
                 found
              </span>
            </p>
            <button
              class="MuiButtonBase-root sc-beqWaB eeHUGz btn-reset-filter css-10d1a0h-MuiButtonBase-root"
              tabindex="0"
              type="button"
            >
              <span
                class="button__start-icon"
              />
              Reset filters
              <span
                class="button__end-icon"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <section
            class="main-list__sections"
          >
            <div
              class="main-list__sections--products"
            >
              <p
                class="text-product__not-found"
              />
            </div>
            <div
              class="main-list__sections--filters"
            >
              <section
                class="main-filter-block"
              >
                <div
                  class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
                >
                  <div
                    class="header-filter"
                  >
                     Sort by  
                  </div>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSuccess MuiInputBase-formControl  css-meedmd-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                      role="button"
                      tabindex="0"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </div>
                    <input
                      aria-hidden="true"
                      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                      tabindex="-1"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-ihdtdm"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
                >
                  <label
                    class="header-filter"
                  >
                    Brand
                  </label>
                </div>
                <div
                  class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
                >
                  <label
                    class="header-filter"
                  >
                    Category
                  </label>
                </div>
                <div>
                  <div
                    class="MuiBox-root css-gmuwbf"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                      height="20px"
                    >
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSuccess MuiInputBase-formControl MuiInputBase-sizeSmall css-kzf2d7-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                          id=":r0:"
                          type="text"
                          value=""
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-ihdtdm"
                          >
                            <span
                              class="notranslate"
                            >
                              ​
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                    <span
                      class="line"
                    >
                       
                    </span>
                    <div
                      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                    >
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSuccess MuiInputBase-formControl MuiInputBase-sizeSmall css-kzf2d7-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                          id=":r1:"
                          type="text"
                          value=""
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-ihdtdm"
                          >
                            <span
                              class="notranslate"
                            >
                              ​
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
                >
                  <label
                    class="header-filter header-filter__name"
                  >
                    Procesor
                  </label>
                </div>
                <div
                  class="show-more-filters"
                >
                  <button
                    class="MuiButtonBase-root sc-beqWaB kPxoob white-shadow css-10d1a0h-MuiButtonBase-root"
                    tabindex="0"
                    type="button"
                    variant="white-shadow"
                  >
                    <span
                      class="button__start-icon"
                    />
                    Show more filters
                    <span
                      class="button__end-icon"
                    />
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </section>
            </div>
          </section>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg main-list css-1oqqzyl-MuiContainer-root"
      >
        <div>
          <div
            aria-hidden="true"
            class="MuiBackdrop-root css-1mgwyik-MuiBackdrop-root"
            style="opacity: 0.7; background-color: white; position: absolute; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
          >
            <span
              class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          class="breadcrumbs__wrapper"
        >
          <a
            class="breadcrumbs__link"
            href="/"
          >
            Home
          </a>
          <a
            class="breadcrumbs__link"
            href="/products"
          >
            Products
          </a>
        </div>
        <div
          class="products-header"
        >
          <p
            class="count-found-product"
          >
            Products 
            <span
              class="title_contrast"
            >
              0
               found
            </span>
          </p>
          <button
            class="MuiButtonBase-root sc-beqWaB eeHUGz btn-reset-filter css-10d1a0h-MuiButtonBase-root"
            tabindex="0"
            type="button"
          >
            <span
              class="button__start-icon"
            />
            Reset filters
            <span
              class="button__end-icon"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <section
          class="main-list__sections"
        >
          <div
            class="main-list__sections--products"
          >
            <p
              class="text-product__not-found"
            />
          </div>
          <div
            class="main-list__sections--filters"
          >
            <section
              class="main-filter-block"
            >
              <div
                class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
              >
                <div
                  class="header-filter"
                >
                   Sort by  
                </div>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSuccess MuiInputBase-formControl  css-meedmd-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
              >
                <label
                  class="header-filter"
                >
                  Brand
                </label>
              </div>
              <div
                class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
              >
                <label
                  class="header-filter"
                >
                  Category
                </label>
              </div>
              <div>
                <div
                  class="MuiBox-root css-gmuwbf"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                    height="20px"
                  >
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSuccess MuiInputBase-formControl MuiInputBase-sizeSmall css-kzf2d7-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                        id=":r0:"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-ihdtdm"
                        >
                          <span
                            class="notranslate"
                          >
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <span
                    class="line"
                  >
                     
                  </span>
                  <div
                    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSuccess MuiInputBase-formControl MuiInputBase-sizeSmall css-kzf2d7-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                        id=":r1:"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-ihdtdm"
                        >
                          <span
                            class="notranslate"
                          >
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
              >
                <label
                  class="header-filter header-filter__name"
                >
                  Procesor
                </label>
              </div>
              <div
                class="show-more-filters"
              >
                <button
                  class="MuiButtonBase-root sc-beqWaB kPxoob white-shadow css-10d1a0h-MuiButtonBase-root"
                  tabindex="0"
                  type="button"
                  variant="white-shadow"
                >
                  <span
                    class="button__start-icon"
                  />
                  Show more filters
                  <span
                    class="button__end-icon"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </section>
          </div>
        </section>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
